ReverseTicTacToe/Logic/Game.cs

```diff
@@ -5,60 +5,308 @@
         public ReverseTicTacToe(int i_BoardDimension, eGameMode i_GameMode)
         {
             //Ofir - since we didnt learn exceptions yet, how should be handle out of range board size in constructor?
-            if (i_BoardDimension < k_GameBoardMinimumDimension || i_BoardDimension > k_GameBoardMaximumDimension)
-                m_BoardDimension = k_DefaultBoardDimension;
+            m_BoardDimension = IsInRange(i_BoardDimension, s_GameBoardMaximumDimension, s_I_GameBoardMinimumDimension)
+                                   ? i_BoardDimension : s_DefaultBoardDimension;
             m_GameMode = i_GameMode;
-            m_Player1Char = k_DefaultPlayer1Char;
-            m_Player2Char = k_DefaultPlayer2Char;
-            m_CurrentTurn = k_DefaultFirstTurn;
+            m_Player1Char = s_DefaultPlayer1Char;
+            m_Player2Char = s_DefaultPlayer2Char;
+            m_CurrentTurn = s_DefaultFirstTurn;
+            m_EmptySlotChar = s_DefaultEmptySlotChar;
             m_GameBoard = new char[m_BoardDimension, m_BoardDimension];
-            m_EmptySlotChar = k_DefaultEmptySlotChar;
+            ClearBoard();
+            m_Player1Score = 0;
+            m_Player2Score = 0;
+            m_GameStatus = eGameStatus.ClearBoard;
+            m_FilledSlotsCounter = 0;
 
-            for(int i = 0; i < m_BoardDimension; i++)
+        }
+
+        public int BoardSize
+        {
+            get
+            {
+                return m_BoardDimension;
+            }
+            set
             {
-                for(int j = 0; j < m_BoardDimension; j++)
+                if(m_GameStatus == eGameStatus.ClearBoard && IsInRange(value, s_I_GameBoardMinimumDimension,
+                       s_GameBoardMaximumDimension ))
+                {
+                    m_BoardDimension = value;
+                    m_GameBoard = new char[m_BoardDimension, m_BoardDimension];
+                    ClearBoard();
+                }
+
+            }
+        }
+
+        public static bool IsInRange(int i_Num, int i_Min, int i_Max)
+        {
+            return i_Num <= i_Max && i_Num >= i_Min;
+        }
+
+        public bool IsSlotIndexWithinBounds(int i_Row, int i_Col)
+        {
+            return (IsInRange(i_Row, 1, m_BoardDimension) && IsInRange(i_Col, 1, m_BoardDimension));
+        }
+
+
+
+        public bool AttemptMove(int i_Row, int i_Col, out eLastMoveStatus o_MoveStatus)
+        {
+            o_MoveStatus = eLastMoveStatus.Good;
+            char charToInsert = (m_CurrentTurn == eTurns.Player1) ? m_Player1Char : m_Player2Char;
+            const bool v_MoveMadeSuccessfully = true;
+            bool moveMade = !v_MoveMadeSuccessfully;
+
+            if(!IsSlotIndexWithinBounds(i_Row, i_Col))
+            {
+                o_MoveStatus = eLastMoveStatus.FailedOutOfBounds;
+            }
+            else if(IsGameOver())
+            {
+                o_MoveStatus = eLastMoveStatus.FailedGameOver;
```

Comment:
I don't see this case where the UI calls AttemptMove when the game is already over

Reply:
i think a method should be built to be as malleable and independent as possible. So even though our CosoleUI will not reach this case, other clients of this class might, if by mistake or not.